yammer-extra/src/main/java/com/arpnetworking/metrics/yammer/Counter.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2015 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.yammer;
+
+import com.arpnetworking.metrics.Metrics;
+import com.yammer.metrics.core.CounterIntercept;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+/**
+ * Represents a Counter that is wrapped to output Arpnetworking Metrics.
+ *
+ * @author Brandon Arp (barp at groupon dot com)
+ */
+@SuppressFBWarnings("NM_SAME_SIMPLE_NAME_AS_SUPERCLASS")
+public class Counter extends CounterIntercept {
+    /**
+     * Public constructor.
+     *
+     * @param name name of the metric
+     * @param lock lock for the metrics reference
+     */
+    public Counter(final String name, final SafeRefLock<Metrics> lock) {
+        _name = name;
+        _lock = lock;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void dec(final long n) {
+        _lock.readLocked(metrics -> {
+                    metrics.resetCounter(_name);
+                    metrics.decrementCounter(_name, n);
+                });
+        super.dec(n);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void inc(final long n) {
+        _lock.readLocked(metrics -> {
+                    metrics.resetCounter(_name);
```

Comment:
This will be a new sample for every increment or decrement call. Can you elaborate, ideally in the class comments on this behavior and why it is desired over the single sample approach. Finally, do we need to consider it a possible future feature to support both paradigms?